draft/planner_test.go

```diff
@@ -0,0 +1,271 @@
+						"version":        "another-version",
+						"version-source": "lowest",
+					},
+				},
+				{
+					Name: "node",
+					Metadata: map[string]interface{}{
+						"version": "other-version",
+					},
+				},
+				{
+					Name: "npm",
+					Metadata: map[string]interface{}{
+						"version":        "some-version",
+						"version-source": "highest",
+					},
+				},
+				{
+					Name: "node",
+					Metadata: map[string]interface{}{
+						"version":        "some-version",
+						"version-source": "highest",
+					},
+				},
+			}, priorities)
+
+			Expect(entry).To(Equal(packit.BuildpackPlanEntry{
+				Name: "node",
+				Metadata: map[string]interface{}{
+					"version":        "some-version",
+					"version-source": "highest",
+				},
+			}))
+
+			Expect(entries).To(Equal([]packit.BuildpackPlanEntry{
+				{
+					Name: "node",
+					Metadata: map[string]interface{}{
+						"version":        "some-version",
+						"version-source": "highest",
+					},
+				},
+				{
+					Name: "node",
+					Metadata: map[string]interface{}{
+						"version":        "another-version",
+						"version-source": "lowest",
+					},
+				},
+				{
+					Name: "node",
+					Metadata: map[string]interface{}{
+						"version": "other-version",
+					},
+				},
+			}))
+		})
+
+		context("the priorities are nil", func() {
+			it("returns the first entry in the filtered map", func() {
+				entry, entries := planner.Resolve("node", []packit.BuildpackPlanEntry{
+					{
+						Name: "node",
+						Metadata: map[string]interface{}{
+							"version":        "another-version",
+							"version-source": "lowest",
+						},
+					},
+					{
+						Name: "node",
+						Metadata: map[string]interface{}{
+							"version": "other-version",
+						},
+					},
+					{
+						Name: "npm",
+						Metadata: map[string]interface{}{
+							"version":        "some-version",
+							"version-source": "highest",
+						},
+					},
+					{
+						Name: "node",
+						Metadata: map[string]interface{}{
+							"version":        "some-version",
+							"version-source": "highest",
+						},
+					},
+				}, nil)
+
+				Expect(entry).To(Equal(packit.BuildpackPlanEntry{
+					Name: "node",
+					Metadata: map[string]interface{}{
+						"version":        "another-version",
+						"version-source": "lowest",
+					},
+				}))
+
+				Expect(entries).To(Equal([]packit.BuildpackPlanEntry{
+					{
+						Name: "node",
+						Metadata: map[string]interface{}{
+							"version":        "another-version",
+							"version-source": "lowest",
+						},
+					},
+					{
+						Name: "node",
+						Metadata: map[string]interface{}{
+							"version": "other-version",
+						},
+					},
+					{
+						Name: "node",
+						Metadata: map[string]interface{}{
+							"version":        "some-version",
+							"version-source": "highest",
+						},
+					},
+				}))
+			})
+		})
+
+		context("there are no enties matching the given name", func() {
+			it("returns the first entry in the filtered map", func() {
```

Comment:
I think you just missed this `it`, I think it should be something like:
```suggestion
			it("returns no entries", func() {
```